Readability.js

```diff
@@ -1689,6 +1695,26 @@ Readability.prototype = {
   },
 
   /**
+   * Clean out elements whose id/class combinations match specific string.
+   *
+   * @param Element
+   * @param RegExp match id/class combination.
+   * @return void
+   **/
+  _cleanMatchedNodes: function(e, regex) {
+    var next = this._getNextNode(e);
```

Comment:
Thanks, @gijsk . Pointed out we need `endOfSearchMarkerNode` here.

But even if we implement `cleanMatchedNodes` like above you mentioned[1], we still need pass all children of `articleContent` node to `cleanMatchedNodes` method to ensure we won't remove any `topCandidate` node, or in this case/bug we will still remove the `<div class="text-wrapper" itemprop="articleBody" id="gigya-share-btns-2_gig_containerParent">` node which is meaningful(we want to keep it here). Does it make sense?

I updated patch for updating `cleanMatchedNodes` method and resort the list. Could you take a look again? Thanks.

[1]: https://github.com/mozilla/readability/pull/319#discussion_r89467125